engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/html/helper/DefaultHtmlContentGenerator.java

```diff
@@ -259,6 +259,11 @@ public String rewrite( final ImageContainer image, ContentItem contentItem ) {
 
   public ContentItem writeImage( final ImageContainer image, final String encoderType, final float quality,
                            final boolean alpha ) throws ContentIOException, IOException {
+
+    if ( dataLocation == null ) {
```

Comment:
From previous commit where logic was added - https://github.com/pentaho/pentaho-reporting/commit/dc5ca2b6bc217f98d9f37da916846d2ae2d2c62b

this check was forgotten - https://github.com/pentaho/pentaho-reporting/blob/ac9b5bf04e0093669c67780e53529470f470ebfa/engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/html/helper/DefaultHtmlContentGenerator.java#L229-L231